crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -106,6 +106,39 @@ fn removed_name(checker: &mut Checker, expr: &Expr, ranged: impl Ranged) {
                 ["airflow", "utils", "dates", "date_range"] => {
                     Some((qualname.to_string(), Replacement::None))
                 }
+                ["airflow", "triggers", "external_task", "TaskStateTrigger"] => {
+                    Some((qualname.to_string(), Replacement::None))
+                }
+                ["airflow", "utils", "file", "TemporaryDirectory"] => {
+                    Some((qualname.to_string(), Replacement::None))
+                }
+                ["airflow", "utils", "file", "mkdirs"] => {
+                    Some((qualname.to_string(), Replacement::None))
+                }
+                ["airflow", "utils", "decorators", "apply_defaults"] => {
+                    Some((qualname.to_string(), Replacement::None))
+                }
```

Comment:
We should mention this can be safely removed (`apply_defaults` is now unconditionally done). It’s probably easier if we add a variant to Replacement (called Message maybe) and do

Replacement::Message(message) => {
    format!("`{deprecated}` is removed in Airflow 3.0; {message}")
}
```

in `impl Violation`.

Reply:
Yep, we'll need something similar for AIR303 and AIR304 as well. or we could probably use this for those 2 🤔

Reply:
This is temporarily removed, but will be added back (probably in the next PR I think?)